lib/pbench/server/cache_manager.py

```diff
@@ -515,25 +554,83 @@ def extract(tarball_path: Path, path: Path) -> Inventory:
 
         Returns:
             An inventory object that mimics an IO[bytes] object while also
-            maintaining a reference to the tarfile TarFile object to be
-            closed later.
+            maintaining a reference to the subprocess.Popen object to be
+            cleaned up later.
 
         Raise:
-            TarballNotFound on failure opening the tarball
             CacheExtractBadPath if the target cannot be extracted
+            TarballUnpackError on other tar-command failures
+            Any exception raised by subprocess.Popen()
+            subprocess.TimeoutExpired if the tar command hangs
         """
-        try:
-            tar = tarfile.open(tarball_path, "r:*")
-        except Exception as exc:
-            raise TarballNotFound(str(tarball_path)) from exc
-        try:
-            stream = tar.extractfile(str(path))
-        except Exception as exc:
-            raise CacheExtractBadPath(tarball_path, path) from exc
-        else:
-            if not stream:
-                raise CacheExtractBadPath(tarball_path, path)
-            return Inventory(stream, tar)
+        tar_path = shutil.which("tar")
+        if tar_path is None:
+            raise TarballUnpackError(
+                tarball_path, "External 'tar' executable not found"
+            )
+
+        # The external tar utility offers better capabilities than the
+        # Standard Library package, so run it in a subprocess:  extract
+        # the target member from the specified tar archive and direct it to
+        # stdout; we expect only one occurrence of the target member, so stop
+        # processing as soon as we find it instead of looking for additional
+        # instances of it later in the archive -- this is a huge savings when
+        # the archive is very large.
+        tar_command = [
+            str(tar_path),
+            "xf",
+            tarball_path,
+            "--to-stdout",
+            "--occurrence=1",
+            path,
+        ]
+        tarproc = subprocess.Popen(
+            tar_command,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
```

Comment:
If we successfully create the subprocess, but fail out before returning an `Inventory` (e.g., on a timeout), we should probably make an attempt to shut things down as gracefully as possible? Basically the `Inventory.close`, perhaps?